src/Domain/DomainCreators/Rectangle.cpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/DomainCreators/Rectangle.hpp"
+
+#include <array>
+#include <unordered_set>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Domain/BlockNeighbor.hpp"
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeVector.hpp"
+
+namespace DomainCreators {
+
+Rectangle::Rectangle(
+    const typename LowerBound::type lower_xy,
+    const typename UpperBound::type upper_xy,
+    const typename IsPeriodicIn::type is_periodic_in_xy,
+    const typename InitialRefinement::type initial_refinement_level_xy,
+    const typename InitialGridPoints::type initial_number_of_grid_points_in_xy,
+    const OptionContext& /*context*/)
+    : lower_xy_(lower_xy),
+      upper_xy_(upper_xy),
+      is_periodic_in_xy_(is_periodic_in_xy),
+      initial_refinement_level_xy_(initial_refinement_level_xy),
+      initial_number_of_grid_points_in_xy_(
+          initial_number_of_grid_points_in_xy) {}
+
+Domain<2, Frame::Inertial> Rectangle::create_domain() const {
+  using affine_map = CoordinateMaps::AffineMap;
```

Comment:
Since this is being used as a type as opposed to a metavariable I'd recommend `using AffineMap`